### PR TITLE
fix: Removed Float64 from SpannerToClrTypeDefaultMappings

### PR DESCRIPTION
**Pull request by Rishabh-V**
This is for initial review from Jon (WIP)

### CODE REVIEW

**Review by jskeet — COMMENTED** [review]
Yes, these new options look right to me.

As discussed:

- We don't want to support converting from Float64 to SpannerNumeric/PgNumeric
- We can infer SpannerNumeric or PgNumeric based on type annotation code

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConversionOptions.cs [hunk 1]
```diff
@@ -174,12 +158,6 @@ private void ValidateAndParseSpannerToClrTypeMappings(string input, SpannerConve
             }
 
             // Check multiples for each type.
-            // Currently, we have Float64 and Date to check.
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConversionOptions.cs:177]: I'm not checking this part massively carefully as I still hope to be able to refactor it. Although there's a bit less need now...

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs [hunk 2]
```diff
@@ -618,15 +612,7 @@ private object ConvertToClrTypeImpl(Value wireValue, System.Type targetClrType,
...
-                bool isFloat64 = TypeCode == TypeCode.Float64;
-                // Check if Float64 is being read as PgNumeric due to user provided type mapping. 
-                bool configuredFloat64AsPgNumeric = isFloat64 && options.Float64ToConfiguredClrType == typeof(PgNumeric); 
-                // For PgNumeric, the value is contained in Value.StringValue.
-                // If the user configures that Float64 should be mapped to PgNumeric CLR type in connection string,
-                // we will need to read Float64 as PgNumeric. The Float64 value is populated in NumberValue of Value.
-                // If the TypeCode is Float64 due to user configured mapping or TypeAnnotationCode is PgNumeric, it is fine.
-                // In all other cases, throw an exception.
-                if (!isPgNumeric && !configuredFloat64AsPgNumeric)
+                if (!isPgNumeric)
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs:615]: Just change to `if (TypeAnnotationCode != TypeAnnotationCode.PgNumeric)` perhaps?

**Review by jskeet — APPROVED** [review]
Much happier about SpannerConversionOptions now :)

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs (hunk 2 above)
Comment (Rishabh-V) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs]: Done.

**Pull request merged**